packages/query-core/src/mutation.ts

```diff
@@ -298,6 +302,8 @@ export class Mutation<
           return {
             ...state,
             data: action.data,
+            failureCount: 0,
+            failureReason: null,
```

Comment:
I think when `loading` is dispatched, we should also reset the `failureCount` and the `failureReason`. At least this is what's happening in `useQuery`:

https://github.com/TanStack/query/blob/04945f38b3f2ff82a597083bf1a36724f973bc7e/packages/query-core/src/query.ts#L518

Reply:
seems like we don't have a test for this behaviour? Could you maybe add one?

Reply:
Ok will do :)